tests/test_controller.py

```diff
@@ -444,28 +443,32 @@ def test_register_fails_on_non_authentication_document(self):
             eq_(INVALID_INTEGRATION_DOCUMENT, response)
 
     def test_register_fails_on_non_matching_id(self):
-        """The request returns an authentication document but its `id`
-        doesn't match the URL it was retrieved from.
-        """
+        # The request returns an authentication document but its `id`
+        # doesn't match the final URL it was retrieved from.
         auth_document = self._auth_document()
-        self.http_client.queue_response(200, content=json.dumps(auth_document))
+        self.http_client.queue_response(
+            200, content=json.dumps(auth_document),
+            url="http://a-different-url/"
+        )
```

Comment:
Of all the tests we had to change to accommodate the new `final_url` checking code, this is the only tricky one. The failure here is that although the final location of the authentication document matches the URL in the registration form submission, it doesn't match the `id` of the authentication document itself.